### ISSUE TITLE
Date is not a constructor

### ISSUE THREAD

**Issue by FezVrasta** [tmpvar/jsdom#1631]
I'm using jsdom with Jest + Enzyme + React.

When I mount a component with Enzyme `mount` after I properly defined `global.window` and `global.document`, I get an error:

```
TypeError: Date is not a constructor

      at DocumentImpl (node_modules/jsdom/lib/jsdom/living/nodes/Document-impl.js:269:83)
      at Object.setup (node_modules/jsdom/lib/jsdom/living/generated/Document.js:702:17)
      at Object.create (node_modules/jsdom/lib/jsdom/living/generated/Document.js:672:10)
      at new Window (node_modules/jsdom/lib/jsdom/browser/Window.js:67:29)
      at Object.<anonymous>.exports.jsdom (node_modules/jsdom/lib/jsdom.js:109:18)
```

I narrowed down the problem to this line of code:
https://github.com/tmpvar/jsdom/blob/master/lib/jsdom/living/nodes/Document-impl.js#L269

Seems like `Date` is `undefiend`... How is it possible?

**Comment by domenic** [comment]
Can you share a minimal example that triggers this problem, without using external libraries like Jest or Enzyme or React?

**Comment by FezVrasta** [comment]
I don't know how to reproduce this problem sorry, I hoped that someone encountered the same problem previously...

**Comment by FezVrasta** [comment]
So, I was using Sinon and I forgot to mention it. By the way, removing Sinon everything works perfectly... So maybe it's something related to Jest + Sinon since they are the only constants in all the discussions I found on the web.

**Comment by mroderick** [comment]
Unrelated to jsdom entirely, but leaving breadcrumbs people running into this issue.

This has been resolved with a [v1.5.2](https://github.com/sinonjs/lolex/releases/tag/v1.5.2) of [sinonjs/lolex](https://github.com/sinonjs/lolex).

If you're using one of the the Sinon `2.0.0-pre` releases, you just need to re-install the lolex dependecy.
